src/rust/engine/src/externs/stdio.rs

```diff
@@ -64,6 +65,31 @@ py_class!(pub class PyStdioRead |py| {
     def fileno(&self) -> PyResult<i32> {
       stdio::get_destination().stdin_as_raw_fd().map_err(|e| PyErr::new::<exc::Exception, _>(py, (e,)))
     }
+
+    def readinto(&self, obj: PyObject) -> PyResult<usize> {
+      let py_buffer = PyBuffer::get(py, &obj)?;
+      let mut buffer = vec![0; py_buffer.len_bytes() as usize];
+      let read = py.allow_threads(|| {
+        stdio::get_destination().read_stdin(&mut buffer)
```

Comment:
Do the types not quite align to allow skipping the middleman with:
```rust
stdio::get_destination().read_stdin(py_buffer.as_mut_slice(py))
```

Reply:
I didn't try it, but `as_mut_slice` gives you back a `&[Cell<T>]`, which I don't think can be treated as a `&[u8]`: only poked individually. Will add a comment.